### ISSUE TITLE
Send a welcome text or email to new subscribers

### ISSUE THREAD

**Issue by moveson** [SplitTime/OpenSplitTime#447]
When someone signs up to receive text or email notifications, we should send a welcome text explaining what they have subscribed for. This increases confidence that the process is working.

**Comment by moveson** [comment]
## Analysis and proposed implementation (2026-04-29)

### Scope

A welcome message is sent **on Subscription creation**, per the subscriber's chosen protocol, only when the protocol is one we control delivery for:

| Protocol | Welcome | Delivered by |
|---|---|---|
| `email` | Yes | Mailgun (existing infrastructure) |
| `sms` | Yes | Pinpoint SMS Voice v2 `send_text_message` (new service) |
| `http`, `https` | No | n/a |

**Out of scope for this issue:** the separate one-time SMS welcome that fires when a user opts in to the SMS program via `/user_settings/sms_messaging` (sample 1 in the new TCR campaign). That's a different concept (per-user, fires on app-side consent, no `Subscription` record exists yet at that point). It can be a follow-up.

### Architecture

1. **`Subscription#after_create_commit :send_welcome`** — single new callback. Dispatches to email or SMS based on `protocol`; no-op for http/https.
2. **`SubscriptionMailer.welcome(subscription).deliver_later`** — new `ActionMailer` subclass alongside existing `ApplicationMailer` / `AdminMailer`. Uses Mailgun (already wired). Effort and Person subscriptions get distinct templates because the natural welcome wording differs ("you'll get updates when Joe passes aid stations" vs "we'll notify you when Joe is registered for an event").
3. **`SmsSubscriptionWelcomeSender.send(subscription)`** — new service that calls `Aws::PinpointSMSVoiceV2::Client#send_text_message` directly with the user's phone, the OST origination number, and the TCR registration's configuration set. Effort-only (SMS for person subscriptions was removed in #1924).

### Message text

**SMS (Effort subscriptions only)** — verbatim TCR-attested sample 2 from the new campaign currently in review (`registration-c797e6734f654e33b61d21b2522b82d1`):

```
OpenSplitTime: You're now subscribed to live progress updates for {effort.full_name} at {effort.event_name}. You'll receive an SMS each time {effort.first_name} passes an aid station. Reply STOP to cancel.
```

**Email — Effort:**
> Subject: You're following {effort.full_name} at {effort.event_name}
>
> You're now subscribed to live progress updates for **{effort.full_name}** at **{effort.event_name}**. You'll receive an email each time they pass an aid station, so you can follow along in real time.
>
> View live results: {results_link}
>
> [unsubscribe link]

**Email — Person:**
> Subject: You're following {person.full_name}
>
> You're now subscribed to follow **{person.full_name}**. We'll send you an email when {person.first_name} is registered for upcoming events on OpenSplitTime, so you can subscribe to live progress updates for those efforts.
>
> [unsubscribe link]

### Compliance / deploy ordering

The SMS welcome wording is **only TCR-attested under the new campaign currently in review**. The active v7 campaign's three samples are all progress notifications. Deploying SMS welcomes against the active campaign before cutover would send untracted text that carriers could flag.

Deploy ordering must be:

1. Build PR (mailer + SMS sender + callback + specs) — can merge anytime
2. **Gate the SMS path behind a credential** (e.g., `aws.sms_welcome_enabled`) — defaults to false
3. Wait for new TCR campaign approval (`registration-c797e6734f654e33b61d21b2522b82d1`, currently `REVIEWING`)
4. Cutover phone `+17626898865` to the new campaign (separate ~5-min outage)
5. Flip the flag to enable SMS welcomes in production

The email path has no TCR exposure and can ship immediately.

### Edge cases worth handling explicitly in tests

- A user re-subscribes after destroying a prior subscription → welcome fires again (acceptable; each fresh Subscription is a new event).
- A user opts out of SMS at the carrier level (`sms_carrier_opted_out_at` set) but admin creates an SMS subscription on their behalf → AWS will silently drop the welcome SMS. Acceptable, but the new `Subscription` record would still get created. The interactor in the SMS sender should short-circuit if `user.sms_carrier_opted_out?` to avoid wasted send.
- An admin (per the still-active #1892 gate) creates an SMS subscription for another user → welcome SMS goes to that user's phone, not the admin's. Confirm the recipient resolution uses `subscription.user.phone`, not `current_user.phone`.
- Mailgun delivery failure → `deliver_later` retries via Solid Queue's retry policy; the Subscription record persists either way.
- PSV2 send failure → swallow the AWS error and surface it via `Rails.error.report(handled: true)`. The Subscription record is already saved; failing the welcome shouldn't roll it back.

### Test plan

- `Subscription` model spec: `after_create_commit` fires the right delegate per protocol; no-op for http/https.
- `SubscriptionMailer` spec: `welcome` for an Effort renders effort-context body; for a Person renders person-context body.
- `SmsSubscriptionWelcomeSender` spec: builds the exact TCR sample-2 wording from a real Effort fixture; calls PSV2 with the right origination number and message body; short-circuits when `user.sms_carrier_opted_out?`.
- System spec (admin-driven, given #1892): admin subscribes a user to an effort by SMS → SMS welcome enqueues; by email → email welcome enqueues.

### Roughly sized

Maybe ~250 lines of code + ~150 lines of specs across 6-8 files. One-day PR scoped tightly, plus the deploy gating + post-approval cutover.

### Cross-references

- Related to #1947 (just closed — app-side SMS opt-out tracking; provides the `sms_carrier_opted_out?` predicate this implementation depends on).
- Blocked on TCR approval of `registration-c797e6734f654e33b61d21b2522b82d1` for the SMS arm only; email arm has no blocker.
- Adjacent to #1892 (admin-only gate on SMS subscribe — most subscriptions will be admin-initiated until that ships).

🤖 Posted by [Claude Code](https://claude.com/claude-code)
